workspaces/arborist/lib/arborist/build-ideal-tree.js

```diff
@@ -1299,7 +1299,7 @@ This is a one-time fix-up, please be patient...
         continue
       }
 
-      const parentEdge = node.parent.edgesOut.get(edge.name)
+      const parentEdge = node.parent.edgesOut.get(edge.name) || node.parent?.edgeOut?.get(edge.name)
```

Comment:
I don't see how this will prevent the exception, let alone fix the bug.

Reply:
> I don't see how this will prevent the exception, let alone fix the bug.

Updated ...